playbooks/openstacksdk-ansible-functional-devstack/run.yaml

```diff
@@ -0,0 +1,19 @@
+- hosts: all
+  become: yes
+  tasks:
+    - name: Clone opnstacksdk
+      git:
+        repo: http://git.openstack.org/openstack/openstacksdk
```

Comment:
We never want to clone the repo directly from git.o.o - zuul should be preparing the git repo state and then it shoud be pushed to the node as part of the base job.

This will result in the change in question not being the thing that is tested.

Reply:
To ensure that this happens - we need to add openstack/openstacksdk and openstack-infra/shade to the openlab tenant, and then add each of them to the required-projects list of the appropriate jobs.

Reply:
it just also because the `openstacksdk` and `shade` have themselves job definitions in `.zuul.yaml` which have a dependencies different with openlab's job structure. if we add them in required-projects, zuul will also try to load the jobs definitions from `openstacksdk` and `shade` and will cause issue. that will cause errors. if we want to support this, we need to import the full dependencies chain of openstacksdk and shade jobs into openlab, that need to do many refactors.

Reply:
Ah - so, that's ok. There is a feature you can use in the zuul main.yaml to tell it to not load any zuul config objects from the repositories:

https://github.com/openstack-infra/project-config/blob/master/zuul/main.yaml#L1428-L1431

Is an example of us using it for ansible/ansible itself (which you might also want to do for the ansible/ansible role - because in the future we will be adding a .zuul.yaml to ansible/ansible itself. ```include: []``` tells zuul "don't load any config objects from this repo" - and allows safely adding shade and openstacksdk to required-projects without the job dependency problem you mention.

I do think it would be cool for us to rework some things to allow loading the jobs too - but I think that's much work - and as you say could impact existing jobs, so let's talk about it later.

Reply:
That is cool, thanks for teaching me 😄 , Ihave tested, it works, will update this PR.